packages/api/lambdas/sf-scheduler.js

```diff
@@ -3,64 +3,65 @@
 const get = require('lodash.get');
 const merge = require('lodash.merge');
 const uuidv4 = require('uuid/v4');
-const { getS3Object, parseS3Uri } = require('@cumulus/common/aws');
+const { getMessageFromTemplate } = require('@cumulus/ingest/queue');
 const { SQS } = require('@cumulus/ingest/aws');
 const { Provider, Collection } = require('../models');
 
-/**
- * Builds a cumulus-compatible message and adds it to the startSF queue
- * startSF queue will then start a stepfunction for the given message
- *
- * @param   {Object} event   - lambda input message
- * @param   {Object} context - lambda context
- * @param   {function} cb    - lambda callback
- */
-function schedule(event, context, cb) {
-  const template = get(event, 'template');
+const buildCumulusMeta = (queueName) => ({
+  execution_name: uuidv4(),
+  queueName
+});
+
+function getProvider(providerId) {
+  if (providerId) {
+    const p = new Provider();
+    return p.get({ id: providerId });
+  }
+  return {};
```

Comment:
One thing I just noticed: before the provider/collection were set on the message as null if not value exists, but you're defaulting to an empty object. I don't think there's anything wrong with that, but is that intentional?

Reply:
this preserves the current functionality:
```
      message.meta.provider = {};	
      message.meta.collection = {};	
      message.meta = merge(message.meta, meta);
```